spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -886,7 +886,14 @@ else if (methodName.equals("close")) {
 
 				// If we get here, we're supposed to shut down.
 				physicalClose();
-				releasePermit();
+				/*
+				 *  Only release a permit if this is a normal close; if the channel is
+				 *  in the list, it means we're closing a cached channel (for which a permit
+				 *  has already been released).
+				 */
+				if (!this.channelList.contains(proxy)) {
+					releasePermit();
+				}
```

Comment:
Yes, it's needed - but as you point out, we didn't have a test case to prove it.

Added.